pkg/store/bucket.go

```diff
@@ -1114,7 +1128,14 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 				s.metrics.chunkSizeBytes.Observe(float64(chunksSize(series.Chunks)))
 			}
 			series.Labels = labelpb.ZLabelsFromPromLabels(lset)
-			if err = srv.Send(storepb.NewSeriesResponse(&series)); err != nil {
+
+			resp := storepb.NewSeriesResponse(&series)
```

Comment:
That isn't quite as easy - `NewSeriesResponse` returns a struct that is made by generated code so we would have to use some kind gogoprotobuf's extension to achieve this unless I am missing something. Given that gogoprotobuf is deprecated and unmaintained, I think it would be better to leave this as is and then work on moving to the protobuf v2 API with the Vitess extensions so that we could use a `sync.Pool` on the querier side as well. IMHO it would feel like a waste of time to hook ourselves even more into gogoprotobuf's custom extensions just to remove all of it a couple of months after that :s

Reply:
Actually, let me try to rework this into something more maintainable

Reply:
<3